kythe/typescript/testdata/tsx.tsx

```diff
@@ -0,0 +1,17 @@
+// Test TypeScript JSX
+
+export {};
+
+function render() {
+  //- @#0"value" defines/binding Value
+  const value = 'value';
+  return (
+    //- @"attr" defines/binding vname("render.div.attr", _, _, _, _)
+    //- @"value" ref Value
+    //- @+3"src" defines/binding vname("render.div.img.src", _, _, _, _)
```

Comment:
I'm not sure this is right.  Look at what the output is to see what these attributes actually do semantically:

http://www.typescriptlang.org/play/?jsx=2#code/GYVwdgxgLglg9mABAJwKZgCauQCgJSIDeAUIiqlCMkjqWYgDwYwBuiAhlFMgLyEvsANiFQBfAHx16jGAFsA5ogDOyCHwHCxiAPSTpDbcxZ7EeANzFRQA

I think we should treat 'src' and 'attr' in the same way we treat keys in objects.

Reply:
What do you mean by that? We currently visit JSX attributes as variable declarations, same as we do property declarations

Reply:
I the attributes effectively become a `{ src: ... }` object literal.  So I think your naming here won't work if there are two of the same tags as siblings, in a layout like

```
<div>
  <img src={a}/>
  <img src={b}/>
</div>
```

I think maybe we need to use anonymous name generation for these, and ignore the tag names entirely.

Reply:
Fixed by creating an anonymous names for all JSX nodes. This is not needed for some nodes, but reduces code complexity since we don't prioritize JSX anyway.